pbs_light.go

```diff
@@ -284,6 +295,52 @@ func auction(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
 		}
 	}
 
+	if pbs_req.SortBids == 1 {
+		// TODO (pbm): the below setting is the default for price granularity
+		priceGranularitySetting := DEFAULT_PRICE_GRANULARITY
+
+		// record bids by ad unit code for sorting
+		code_bids := make(map[string]pbs.PBSBidSlice)
+		for _, bid := range pbs_resp.Bids {
+			code_bids[bid.AdUnitCode] = append(code_bids[bid.AdUnitCode], bid)
+		}
+
+		// loop through ad units to find top bid
+		for _, unit := range pbs_req.AdUnits {
+			bar := code_bids[unit.Code]
+
+			if len(bar) == 0 {
+				if glog.V(1) {
+					glog.Infof("No bids for ad unit '%s'", unit.Code)
+				}
+				continue
+			}
+			sort.Sort(bar)
+
+			// after sorting we need to add the ad targeting keywords
+			for i, bid := range bar {
+				priceBucketStringMap := pbs.GetPriceBucketString(bid.Price)
+				roundedCpm := priceBucketStringMap[priceGranularitySetting]
+
+				hbPbBidderKey := "hb_pb_" + bid.BidderCode
+				hbBidderBidderKey := "hb_bidder_" + bid.BidderCode
+				hbCacheIdBidderKey := "hb_cache_id_" + bid.BidderCode
+				pbs_kvs := map[string]string{
+					hbPbBidderKey[:min(len(hbPbBidderKey), 20)]:      roundedCpm,
```

Comment:
or a config flag like "this is dfp"? It will break things if we don't truncate, right?